### PR TITLE
defect-90:Resize: Add Network - data is incorrect

### PR DESCRIPTION
**Pull request by rahulray19**

### CODE REVIEW

File: src/main/groovy/com/morpheusdata/xen/XenserverProvisionProvider.groovy [hunk 1]
```diff
@@ -1808,8 +1809,8 @@ class XenserverProvisionProvider extends AbstractProvisionProvider implements Wo
 
 	def buildNetworkInterface(server, networkResults, newNetwork, newIndex, index) {
```
Comment (ddeyoung) [inline src/main/groovy/com/morpheusdata/xen/XenserverProvisionProvider.groovy:1810]: having index and new index arguments here is confusing. Only one index is needed, so remove the last argument `index` that isn't used.

File: src/main/groovy/com/morpheusdata/xen/XenserverProvisionProvider.groovy [hunk 2]
```diff
@@ -1742,6 +1742,7 @@ class XenserverProvisionProvider extends AbstractProvisionProvider implements Wo
 					if (networkResults.success == true) {
 						def newInterface = buildNetworkInterface(computeServer, networkResults, newNetwork, newIndex, index)
 						newInterface.uniqueId = isWorkload ? "morpheus-nic-${instanceId}-${workload.id}-${newIndex}" : "morpheus-nic-${server.id}-${newIndex}"
+						newInterface.primaryInterface = networkAdd?.network?.isPrimary ? true : false
```
Comment (ddeyoung) [inline src/main/groovy/com/morpheusdata/xen/XenserverProvisionProvider.groovy:1745]: I don't think `isPrimary` exists and `networkAdd.network` is the network associated to the interface, so it wouldn't have information to determine if the interface should be primary. 

A primary interface can't be removed or a new primary interface added, so `newInterface.primaryInterface` will always be false.

**Review by ddeyoung — CHANGES_REQUESTED** [review]

File: src/main/groovy/com/morpheusdata/xen/XenserverProvisionProvider.groovy (hunk 2 above)
Comment (rahulray19) [reply src/main/groovy/com/morpheusdata/xen/XenserverProvisionProvider.groovy]: review comment changes added

File: src/main/groovy/com/morpheusdata/xen/XenserverProvisionProvider.groovy (hunk 1 above)
Comment (rahulray19) [reply src/main/groovy/com/morpheusdata/xen/XenserverProvisionProvider.groovy]: unused index param removed as suggested

**Review by ddeyoung — APPROVED** [review]

**Pull request merged**